src/hotspot/share/opto/superword.hpp

```diff
@@ -668,12 +659,66 @@ class SuperWord : public ResourceObj {
   // Find the longest type in def-use chain for packed nodes, and then compute the max vector size.
   int max_vector_size_in_def_use_chain(Node* n);
 
-  static LoadNode::ControlDependency control_dependency(Node_List* p);
   // Alignment within a vector memory reference
   int memory_alignment(MemNode* s, int iv_adjust);
-  // Ensure that the main loop vectors are aligned by adjusting the pre loop limit.
-  void determine_mem_ref_and_aw_for_main_loop_alignment();
-  void adjust_pre_loop_limit_to_align_main_loop_vectors();
+
+  bool vtransform() const;
+};
+
+// Facility class that builds a VTransformGraph from a SuperWord PackSet.
+class SuperWordVTransformBuilder : public StackObj {
```

Comment:
Note: building the graph used to be done directly in `PacksetGraph`. This is the translation from `SuperWord` to `VTransform`. In the future, we could have other vectorization approaches that construct a `VTransform` with something else than `SuperWord`.